### PR TITLE
adding common-cache module and some classes in authorization-impl module

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**
I use guava LoadingCache for temporary saving registration data, also added POJO class for storing : aut_code, phone_number , registration_session_uuid, other... , also added AuthorizationCacheService - it is descendant of CacheService abstract class from common-cache module

### CODE REVIEW

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/AuthorizationCacheService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * @author AlexKazmirchuk
+ * @since 20.03.17.
+ */
+@Service
+public class AuthorizationCacheService extends CacheService<RegistrationCacheData> {
+
+    @Override
+    public RegistrationCacheData loadFromSource(String key) {
+        return null;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/AuthorizationCacheService.java:15]: @AlexKazmirchuk Guava does not allow null values! You will get exception.

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/RegistrationCacheData.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+/**
+ * @author AlexKazmirchuk
+ * @since 20.03.17.
+ */
+
+public class RegistrationCacheData implements Serializable {
+
+    private static final long serialVersionUID = -7589939630840486542L;
+
+    private String registrationRequestUuid;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/RegistrationCacheData.java]: @AlexKazmirchuk No need to store this field here. This is a key in cache.

File: softgroup/services/authorization/authorization-imp/src/test/java/com/softgroup/authorization/impl/tests/AuthorizationCacheServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * @author AlexKazmirchuk
+ * @since 20.03.17.
+ */
+
+public class AuthorizationCacheServiceTest {
+
+
+    @Test
+    public void saveAndGetMethodTest(){
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/test/java/com/softgroup/authorization/impl/tests/AuthorizationCacheServiceTest.java]: @AlexKazmirchuk Split into different methods.

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/AuthorizationCacheService.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/AuthorizationCacheService.java:15]: @Dominys  I know. And I catch this exception in get method of abstract class CacheService [(link)](https://github.com/AlexKazmirchuk/cources.server/blob/06f0752633c2ee2b4dd09599fd3d47920a2e322f/softgroup/basis/common-cache/src/main/java/com/softgroup/common/cache/impl/CacheService.java#L46-L57)

```java

public abstract class CacheService<T> implements Cacheable<T> {

...
// some other code
...

@Override
   public  T get(String key){
        if (key == null){
            return null;
        }
        try {
            return cache.get(key);
        } catch (ExecutionException e) {
            return null;
        } catch (CacheLoader.InvalidCacheLoadException ex){
            return null;
        }
    }
```

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/RegistrationCacheData.java (hunk 2 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/cache/RegistrationCacheData.java]: @Dominys done! deleted this redundant field

File: softgroup/services/authorization/authorization-imp/src/test/java/com/softgroup/authorization/impl/tests/AuthorizationCacheServiceTest.java (hunk 3 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-imp/src/test/java/com/softgroup/authorization/impl/tests/AuthorizationCacheServiceTest.java]: @Dominys splitted !

**Review by Dominys — APPROVED** [review]

**Pull request merged**